src/main/java/crawlercommons/sitemaps/SiteMapIndexSAXHandler.java

```diff
@@ -0,0 +1,104 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package crawlercommons.sitemaps;
+
+import static crawlercommons.sitemaps.SiteMapParser.LOG;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Date;
+import java.util.LinkedList;
+
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+
+import crawlercommons.sitemaps.AbstractSiteMap.SitemapType;
+
+/**
+ * Parse XML that contains a Sitemap Index. Example Sitemap Index:
+ * 
+ * <?xml version="1.0" encoding="UTF-8"?> <sitemapindex
+ * xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"> <sitemap>
+ * <loc>http://www.example.com/sitemap1.xml.gz</loc>
+ * <lastmod>2004-10-01T18:23:17+00:00</lastmod> </sitemap> <sitemap>
+ * <loc>http://www.example.com/sitemap2.xml.gz</loc>
+ * <lastmod>2005-01-01</lastmod> </sitemap> </sitemapindex>
+ * 
+ */
+class SiteMapIndexSAXHandler extends AbstractSiteMapSAXHandler {
+
+    private SiteMapIndex sitemap;
+    private URL loc;
+    private Date lastMod;
+    private int i = 0;
+
+    SiteMapIndexSAXHandler(URL url, LinkedList<String> elementStack, boolean strict) {
+        super(elementStack, strict);
+        sitemap = new SiteMapIndex(url);
+        sitemap.setType(SitemapType.INDEX);
+    }
+
+    public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+    }
+
+    public void endElement(String uri, String localName, String qName) throws SAXException {
+        if ("sitemap".equals(currentElement())) {
+            maybeAddSiteMap();
+        } else if ("sitemapindex".equals(currentElement())) {
+            sitemap.setProcessed(true);
+        }
+    }
+
+    public void characters(char[] ch, int start, int length) throws SAXException {
+        String localName = super.currentElement();
+        String value = String.valueOf(ch, start, length);
+        if ("loc".equals(localName)) {
+            try {
+                loc = new URL(value);
+            } catch (MalformedURLException e) {
+                LOG.trace("Don't create an entry with a bad URL", e);
+                LOG.debug("Bad url: [{}]", loc);
+            }
+        } else if ("lastmod".equals(localName)) {
+            lastMod = SiteMap.convertToDate(value);
+            ;
+        }
+    }
+
+    public AbstractSiteMap getSiteMap() {
+        return sitemap;
+    }
+
+    private void maybeAddSiteMap() {
+        if (loc != null) {
+            SiteMap s = new SiteMap(loc, lastMod);
+            sitemap.addSitemap(s);
+            LOG.debug("  {}. {}", (i + 1), s);
+        }
+        loc = null;
+        lastMod = null;
+    }
+
+    public void error(SAXParseException e) throws SAXException {
```

Comment:
I can only second guess what @matt-deboer intended. It could be to make sure the loc and lastmod fields are reset to null, however a new instance of the handler class would be created for a  new doc anyway. It could help with the "handle broken sitemaps" if the exception occurred between the discovery of the location and the next sitemap or sitemapindex tag.